node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/configserver/noderepository/RealNodeRepository.java

```diff
@@ -93,14 +93,20 @@ public void addNodes(List<AddNode> nodes) {
                                     node -> new Acl.Node(node.hostname, node.ipAddress),
                                     Collectors.toSet())));
 
+            // Group trusted networks by container hostname that trusts them
+            Map<String, Set<String>> trustedNetworks = response.trustedNetworks.stream()
+                     .collect(Collectors.groupingBy(GetAclResponse.Network::getTrustedBy,
+                                                    Collectors.mapping(node -> node.network, Collectors.toSet())));
+
 
             // For each hostname create an ACL
-            return Stream.of(trustedNodes.keySet(), trustedPorts.keySet())
-                    .flatMap(Set::stream)
-                    .distinct()
-                    .collect(Collectors.toMap(
-                            Function.identity(),
-                            hostname -> new Acl(trustedPorts.get(hostname), trustedNodes.get(hostname))));
+            return Stream.of(trustedNodes.keySet(), trustedPorts.keySet(), trustedNetworks.keySet())
+                         .flatMap(Set::stream)
+                         .distinct()
+                         .collect(Collectors.toMap(
+                                 Function.identity(),
+                                 hostname -> new Acl(trustedPorts.get(hostname), trustedNodes.get(hostname),
```

Comment:
This will break with the change in the `Acl` constructor since some of those `Map::get`s will return a `null`?

Reply:
Thanks, will fix.